lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -69,6 +69,16 @@
               False in the inventory config file which will allow 403 errors to be gracefully skipped.
           type: bool
           default: True
+        use_legacy_script_group_name_sanitization:
+          description:
+            - By default this plugin is using a general group name sanitization to create safe and usable group names for use in Ansible.
+              This toggle allows those migration from the old ec2.py inventory script that want to continue using the old sanitization to do so.
```

Comment:
> This toggle allows those migration from the old ec2.py

just not for region names.

Reply:
can you point out where region names get added in both script and plugin?

Reply:
### Script

adding ami_id, does to_safe, and then push

https://github.com/ansible/ansible/blob/d4bd01c7acaa380b0b011822684e8dbc4d1fb251/contrib/inventory/ec2.py#L983-L984

region, does push_group

https://github.com/ansible/ansible/blob/d4bd01c7acaa380b0b011822684e8dbc4d1fb251/contrib/inventory/ec2.py#L1132

Also see https://github.com/ansible/ansible/pull/53458#issuecomment-470648367, it's hard to prove a negative, there's no substitute for just running it.

I only mentioned this because of the wording of the documentation here. I am planning to go the route s-hertel mentioned.

### Plugin

Region names don't get added in the plugin by default. It is a user-defined entry in `keyed_groups`, as in my above examples.

Reply:
possible rewording for clarity:
"This toggle allows those migration from the old ec2.py and matches the sanitization of groups when the script's `replace_dash_in_groups` option is set to False. To replicate behavior of `replace_dash_in_groups = True`, you will need to replace hyphens with underscores via the `regex_replace` filter for the affected groups."

Reply:
that won't remove auto added groups with - just constructed ones, but good rephrase in general

Reply:
updated

Reply:
does aws_ec2 have any auto added groups? I am not aware of any.

Reply:
if none , then i don't understand your complaint, if all are added via compose you already control sanitization

Reply:
The change you are making here is allowing `-` to be returned. We can replace `-` with `_` with what we control, but we _cannot_ replace `_` with `-` after the plugin has already sanitized them - which is the current state of `devel`.

This toggle does not replicate the ec2.py behavior, because that replaced dashes inconsistently. The simple explanation of the functionality of _this_ toggle is "dashes are not replaced". For compatibility, the user is on their own. The reason this is related to compatibility is that it's theoretically impossible to have compatibility until we get something in `devel` that does not replace the dashes.

Reply:
you can still replace them yourself, as @s-hertel shows you above, i'm not sure we want to make a 'toggle per possible path' when it is already possible to control this

Reply:
yes, that's fine, the toggle per-keyed-group would been a different solution for this, but this PR is sufficient.